test/query-tests/address-entity.test.ts

```diff
@@ -0,0 +1,43 @@
+import { AppDataSource } from '../../src/data-source';
+import { User } from '../../src/entity/User';
+import { createMochaUserEntity, mochaUser } from '../mocha-user';
+import { Address } from '../../src/entity/Address';
+import { faker } from '@faker-js/faker';
+import { expect } from 'chai';
+
+function createRandomAddress(user: User) {
+  const address = new Address();
+  address.cep = faker.address.zipCode();
+  address.street = faker.address.street();
+  address.streetNumber = faker.address.buildingNumber();
+  address.complement = faker.address.secondaryAddress();
+  address.neighborhood = faker.address.county();
+  address.city = faker.address.cityName();
+  address.state = faker.address.state();
+  address.user = user;
+
+  return address;
+}
+
+describe('Address Entity Test', () => {
+  afterEach(async () => {
+    await AppDataSource.getRepository(Address).delete({});
+    await AppDataSource.getRepository(User).delete({ email: mochaUser.email });
+  });
+
+  it('Should bring two addresses vinculated to a temporary user.', async () => {
+    const testUser = await createMochaUserEntity();
+    const createdUser = await AppDataSource.getRepository(User).save(testUser);
+
+    const addresses = [createRandomAddress(createdUser), createRandomAddress(createdUser)];
+
+    await AppDataSource.getRepository(Address).save(addresses);
+    const addressesInDatabase = await AppDataSource.getRepository(Address).find({ relations: { user: true } });
+
+    expect(addresses.length).to.be.eq(addressesInDatabase.length);
+
+    for (let index = 0; index < addresses.length; index++) {
+      expect(addressesInDatabase[index]).to.be.deep.eq(addresses[index]);
+    }
+  });
+});
```

Comment:
O que vc está testando nesse teste 🤔

Reply:
O exercício pediu pra fazer um teste de inclusão de dois endereços, não pediu pra fazer um teste unitário, porém fiz assim pra cumprir o pedido.

Reply:
Mas esse teste unitário está testando apenas se o TypeORM funciona como esperado?

Reply:
Não, está testando se a entidade Address está funcionando. O exercício pede o seguinte:

 - Make sure the relation is working by running some test code and adding a user with at least 2 addresses. Check TablePlus to see the table sctructure working as above.

Estou ciente que não é um teste unitário, porém como não tinha "local" pra colocar esse "test code" pedido, coloquei como teste unitário.

Reply:
Certo, nesse caso não parece valer a pena commitar esse teste, certo? Porque ele está basicamente testando se o TypeORM funciona. O que queremos subir como código de teste no nosso fonte é o que nos ajuda a testar as features que fazemos, não códigos que testem bibliotecas.

Talvez esse código tenha sido útil pra vc saber se a tabela foi criada e se os valores estão inseridos com o a FK correta, então a produção dele é justa. Porém pra ser acrescentado no código fonte, remoto, não parece ter muito valor. Concorda?

Reply:
Com certeza, vou subir um commit removendo esse código.